pkg/controller/batch/batch_controller.go

```diff
@@ -0,0 +1,373 @@
+		}
+		if updated {
+			return reconcile.Result{}, nil
+		}
+	}
+
+	// We update the phase separately to the spec as the status update is ignored when in the update mutate function
+	return reconcile.Result{}, r.UpdatePhase(v2.BatchInitialized, nil)
+}
+
+func (r *batchResource) execute() (reconcile.Result, error) {
+	batch := r.instance
+	infinispan := &v1.Infinispan{}
+	if result, err := kube.LookupResource(batch.Spec.Cluster, batch.Namespace, infinispan, r.client, log); result != nil {
+		return *result, err
+	}
+
+	url, err := connectionUrl(infinispan)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	cliArgs := fmt.Sprintf("--connect '%s' --trustall --file '%s/%s'", url, BatchVolumeRoot, BatchFilename)
+
+	labels := batchLabels(batch.Name)
+	infinispan.AddLabelsForPods(labels)
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      batch.Name,
+			Namespace: batch.Namespace,
+		},
+		Spec: batchv1.JobSpec{
+			BackoffLimit: pointer.Int32Ptr(0),
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Name:    batch.Name,
+						Image:   infinispan.ImageName(),
+						Command: []string{"/opt/infinispan/bin/cli.sh", cliArgs},
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      BatchVolumeName,
+								MountPath: BatchVolumeRoot,
+							},
+						},
+					}},
+					RestartPolicy: corev1.RestartPolicyNever,
+					Volumes: []corev1.Volume{{
+						Name: BatchVolumeName,
+						VolumeSource: corev1.VolumeSource{
+							ConfigMap: &corev1.ConfigMapVolumeSource{
+								LocalObjectReference: corev1.LocalObjectReference{Name: *batch.Spec.ConfigMap},
+							},
+						},
+					}},
+				},
+			},
+		},
+	}
+
+	_, err = controllerutil.CreateOrUpdate(ctx, r.client, job, func() error {
+		return controllerutil.SetControllerReference(batch, job, r.scheme)
+	})
+
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("Unable to create batch job '%s': %w", batch.Name, err)
+	}
+	return reconcile.Result{}, r.UpdatePhase(v2.BatchRunning, nil)
+}
+
+func (r *batchResource) waitToComplete() (reconcile.Result, error) {
+	batch := r.instance
+	job := &batchv1.Job{}
+	if result, err := kube.LookupResource(batch.Name, batch.Namespace, job, r.client, log); result != nil {
+		return *result, err
+	}
+
+	status := job.Status
+	if status.Succeeded > 0 {
+		return reconcile.Result{}, r.UpdatePhase(v2.BatchSucceeded, nil)
+	}
+
+	if status.Failed > 0 {
+		numConditions := len(status.Conditions)
+		if numConditions > 0 {
+			condition := status.Conditions[numConditions-1]
+
+			if condition.Type == batchv1.JobFailed {
+				var reason string
+				podName, err := r.getJobPodName(batch.Name, batch.Namespace)
+				if err != nil {
+					reason = err.Error()
+				} else {
+					reason, err = kubernetes.Logs(podName, batch.Namespace)
```

Comment:
I know that's not related to the Operator code, but could we have more reliable reason message. It's not informative, which command is failed and why some extra signs added
`"\e[0;31mCommandLineParserException: Command and/or sub-command is not valid!\e[0;39m\n"`

Reply:
Agreed. I have created [ISPN-12752](https://issues.redhat.com/browse/ISPN-12752) to address this.